pkg/reader/ccip.go

```diff
@@ -582,6 +585,23 @@ func (r *ccipChainReader) getSourceChainsConfig(
 			if err != nil {
 				return fmt.Errorf("failed to get source chain config: %w", err)
 			}
+
+			// This may happen due to some sort of regression in the codec that unmarshals
+			// chain data -> go struct.
+			if len(resp.OnRamp) == 0 {
+				return fmt.Errorf(
+					"onRamp misconfigured/didn't unmarshal for chain %d: %x",
+					chainSel,
+					resp.OnRamp,
+				)
+			}
+
+			if !resp.IsEnabled {
```

Comment:
This is actually a regression, we had it in `Sync` but seems like it was deleted.